packages/server/src/adapters/zodFileSchema.ts

```diff
@@ -0,0 +1,76 @@
+import type { Readable } from 'stream';
+import z from 'zod';
+
+function isObject(value: unknown): value is Record<string, unknown> {
+  // check that value is object
+  return !!value && !Array.isArray(value) && typeof value === 'object';
+}
+
+const readableEsqueSchema = z.custom<Readable>((value) => {
+  // check if file is is a Readable without importing Readable
+  if (!isObject(value)) {
+    return false;
+  }
+  if (typeof (value as unknown as Readable).on === 'function') {
+    return true;
+  }
+
+  return false;
+});
+
+/**
+ * This is a hack to check if a value is a File in the browser
```

Comment:
`File` is defined in browser so you could just do `instanceof` no? Is this supposed to be "Nodejs"?